lib/ood_core/job/adapters/torque.rb

```diff
@@ -201,7 +201,12 @@ def info_where_owner(owner)
         # @see Adapter#info
         def info(id)
           id = id.to_s
-          parse_job_info(*@pbs.get_job(id).flatten)
+
+          result = @pbs.get_job(id)
+
+          return parse_job_info(*result.flatten) if result.keys.length == 1
+
+          parse_job_array(id, result)
```

Comment:
I thought that `@pbs.get_job(id)` would return either an array of hashes or a single Hash. And that it would return an Array of hashes for a job array. Thus the original implementation assumed a single job was being returned for `info(id)` call i.e. `info("10219837.oak-batch.osc.edu")` would return below:

```
irb(main):009:0> {"10219837.oak-batch.osc.edu" => { :Job_Name => "CFD_Solver" } }.flatten
=> ["10219837.oak-batch.osc.edu", {:Job_Name=>"CFD_Solver"}]
```

And then `*["10219837.oak-batch.osc.edu", {:Job_Name=>"CFD_Solver"}]` makes that two arguments, the id `"10219837.oak-batch.osc.edu"` and the hash `{:Job_Name=>"CFD_Solver"}`.

So I would expect this code to crash, as I would expect `get_job` to return an array if it were a job array, and arrays do not respond to `keys` method. Is this correct?

Also, is it a job array only if the array size is > 1? Or is there any situation where you have a job array where get_job returns only 1 job? If you reliably only get array size > 1 if there is a job array, then you can coerce result into an array using the refinement here https://github.com/OSC/ood_core/blob/d21a1d62a41b1a3e6d05a28cfe1100bb931597f9/lib/ood_core/refinements/array_extensions.rb

```ruby
result = Array.wrap(@pbs.get_job(id))
```

Then the info implementation is simplified to:

```ruby
def info(id)
  id = id.to_s
  result = Array.wrap(@pbs.get_job(id))
  parse_job_info(id, result.first.flatten.last, result.drop(1).map { |k,v| { id: k, status: STATE_MAP.fetch(value[:job_state], :undetermined) } })
```

Or I guess more readable:

```ruby
def child_task_statuses(child_hashes) # what is a good name for the arg child_hashes?
  child_hashes.map do |key, value|
    id: k, status: STATE_MAP.fetch(value[:job_state], :undetermined) } }
  end
end

def info(id)
  id = id.to_s
  result = Array.wrap(@pbs.get_job(id))
  parse_job_info(id, result.first.flatten.last, child_task_statuses(result.drop(1)))